### PR TITLE
Soil amendment schedules 1: `TillageSchedule` and `TillageEvent`

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR contains the `TillageSchedule` and `TillageEvent` modules, which define how tillage applications are scheduled for a field.

- `TillageSchedule` takes all the same parameters as `Schedule`, plus tillage depths, incorporation fractions, and mixing fractions.
- Input is validated with `_validate_tillage_parameters()` and `_validate_pattern_parameters()` (implemented in `Schedule`). `_validate_tillage_parameters()` is supplemented with `_validate_depths()`  and `_validate_fractions()`.
- Contains `generate_tilling_events()` which generates a full list of all tillage application parameters that should be applied to the Field.
- All methods in `TillageSchedule` are tested in `test_amendment_schedule.py`.
- `events.py` now contains `TillageEvent`, a subclass of `Event` which contains the time, depth, incorporation and mixing efficiencies of a single tillage application.

Note: the way this class uses inheritance may not be the best/most efficient way to do so, but will be addressed by https://github.com/RuminantFarmSystems/MASM/issues/561

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/manager/tillage_schedule.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        all_tilling_days = self.days * (self.pattern_repeat + 1)
+        all_tillage_depths = self.tillage_depths * (self.pattern_repeat + 1)
+        all_incorporation_fractions = self.incorporation_fractions * (self.pattern_repeat + 1)
+        all_mixing_fractions = self.mixing_fractions * (self.pattern_repeat + 1)
+        all_tillage_events = list(zip(all_tillage_depths, all_incorporation_fractions, all_mixing_fractions,
+                                      all_tilling_years, all_tilling_days))
+
+        tillage_events = []
+        for event in all_tillage_events:
+            new_tillage_event = TillageEvent(event[0], event[1], event[2], event[3], event[4])
```
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/manager/tillage_schedule.py]:
```suggestion
            new_tillage_event = TillageEvent(tillage_depth = event[0], incorporation_fraction = event[1], mixing_fraction = event[2], year = event[3], day = event[4])
```

Comment (JoeWaddell) [reply SC_redesign/Crop_and_Soil/manager/tillage_schedule.py]: Apologies for funky formatting, commenting in VSCode - but naming the args here might help with code clarity.

File: SC_redesign/Crop_and_Soil/tests_SC/manager_tests/test_tillage_schedule.py
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/tests_SC/manager_tests/test_tillage_schedule.py:1]: A problem on my end, potentially, but I couldn't get VSCode to happily allow me to run the new tests in the sidebar: so I ran it directly after adding it to the Python> testing:Cwd options. However, it gives a couple warnings. Is this a known issue for the time being?

image

**Review by JoeWaddell — APPROVED** [review]
This looks good overall! Left a question on the tests, but those look like nice coverage.

File: SC_redesign/Crop_and_Soil/tests_SC/manager_tests/test_tillage_schedule.py
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/tests_SC/manager_tests/test_tillage_schedule.py:1]: Yes, it is

File: SC_redesign/Crop_and_Soil/manager/tillage_schedule.py (hunk 1 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/manager/tillage_schedule.py]: I agree, good call

**Review by matthew7838 — APPROVED** [review]
Seems good to me, have nothing to comment on this one. Detailed tests.

**Pull request merged**